File: training-bits/friends/judge.cpp
#include "judge.h"
#include <cstdio>

void judge::run(iFriends *solution) {
    int n, a, b;

    scanf("%d", &n);
    for(int i = 0; i < n; i++) {
        scanf("%d %d", &a, &b);
        solution->addFriend(a, b);
    }

    scanf("%d", &n);
    for(int i = 0; i < n; i++) {
        scanf("%d %d", &a, &b);
        printf("%d\n", solution->haveCommonFriend(a, b) ? 1 : 0);
    }
}

File: src/main/java/bits/friends/Judge.java
package bits.friends;

import java.io.IOException;
import java.util.Scanner;

public class Judge {
	public static void run(Friends solution) throws IOException {
		Scanner sc = new Scanner(System.in);
		int add = sc.nextInt();

		for(int i = 0; i < add; i++) {
			int a = sc.nextInt();
			int b = sc.nextInt();
			solution.addFriend(a, b);
		}

		int check = sc.nextInt();
		for(int i = 0; i < check; i++) {
			int a = sc.nextInt();
			int b = sc.nextInt();
			System.out.println(solution.haveCommonFriend(a, b) ? 1 : 0);
		}
		
		sc.close();
	}
}
